### ISSUE TITLE
Help: Overriding SPACE for music player and a11y

### ISSUE THREAD

**Issue by enjikaka** [a11yproject/a11yproject.com#424]
In a music player, SPACE is used to pause/play the currently playing item in the player. However, a11y features might be ruined by this. (If we catch SPACE, users might fail to mark checkboxes which require SPACE and don't listen to ENTER. See: http://webaim.org/techniques/keyboard/ ).

Have anyone had this issue before and discovered a good solution?

**Comment by scottaohara** [comment]
I'm not sure what your issue is here?  

Are you creating a custom audio player that somehow has check boxes inside of it?

As you mentioned, <kbd>space</kbd> is used to pause/play an audio source....but only if keyboard focus is actually on the audio player.  Hitting <kbd>space</kbd> anywhere else in a document will not start/stop an audio player, it will instead do whatever it is that <kbd>space</kbd> is supposed to do in that instance.  Whether that be scrolling the web document down, or if focused on a checkbox, checking/unchecking it.

I suppose if you've built a custom script that globally hijacks all instances of <kbd>space</kbd> to control an audio player, so that a user doesn't need to be focused on an audio player to control it, then I can see why you're having an issue.  But in this instance, what you'd need to do is updated your function to only attach keypress of <kbd>space</kbd> when no other element is focused.  So, inputs, textareas, links...anything where a user would expect <kbd>space</kbd> to work, focusing on those elements would negate the global audio control function.

So...that was a very long response to what I'm guessing your dealing with now?  Let me know if any of that helped.

**Comment by enjikaka** [comment]
> you'd need to do is updated your function to only attach keypress of space when no other element is focused

That's basically what we've planed to do. Do you think that's a11y-OK?

**Comment by scottaohara** [comment]
personally i think you **HAVE** to do that, since by your response, I'm assuming you've hijacked the space key completely to run the audio component and, as you originally mentioned, have broken expected functionality for any other element in the document that would otherwise use the space key.

**Comment by davatron5000** [comment]
Hopefully a good resolution was met here. I'm going to close this issue. 

In the future let's try to keep help requests on a forum like Stack Overflow and limit Github issues to project-related problems/additions/etc.

Thanks.
